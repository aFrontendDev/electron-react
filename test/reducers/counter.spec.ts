import counterReducer from '../../app/reducers/counter';
import {
  INCREMENT_COUNTER,
  DECREMENT_COUNTER
} from '../../app/types/counter.type';

describe('reducers', () => {
  describe('counter', () => {
    it('should handle initial state', () => {
      expect(counterReducer(undefined, {})).toMatchSnapshot();
    });

    it('should handle INCREMENT_COUNTER', () => {
      expect(counterReducer(1, { type: INCREMENT_COUNTER })).toMatchSnapshot();
    });

    it('should handle DECREMENT_COUNTER', () => {
      expect(counterReducer(1, { type: DECREMENT_COUNTER })).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(counterReducer(1, { type: 'unknown' })).toMatchSnapshot();
    });
  });
});
