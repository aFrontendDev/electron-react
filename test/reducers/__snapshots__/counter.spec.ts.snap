// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers counter should handle DECREMENT_COUNTER 1`] = `
Object {
  "count": NaN,
}
`;

exports[`reducers counter should handle INCREMENT_COUNTER 1`] = `
Object {
  "count": NaN,
}
`;

exports[`reducers counter should handle initial state 1`] = `
Object {
  "count": 0,
}
`;

exports[`reducers counter should handle unknown action type 1`] = `1`;
